src/main/kotlin/com/demonwav/mcdev/i18n/translations/Translation.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n.translations
+
+import com.demonwav.mcdev.i18n.reference.I18nReference
+import com.demonwav.mcdev.i18n.translations.identifiers.LiteralTranslationIdentifier
+import com.demonwav.mcdev.i18n.translations.identifiers.ReferenceTranslationIdentifier
+import com.google.common.collect.ImmutableList
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.editor.FoldingGroup
+import com.intellij.openapi.util.TextRange
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiExpressionList
+import com.intellij.psi.PsiReference
+import com.intellij.psi.PsiReferenceProvider
+import com.intellij.psi.PsiReferenceRegistrar
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.ProcessingContext
+import java.util.ArrayList
+
+data class Translation(val foldingElement: PsiElement?,
+                       val referenceElement: PsiElement?,
+                       val key: String,
+                       val varKey: String,
+                       val text: String?,
+                       val formattingError: Boolean = false,
+                       val containsVariable: Boolean = false) {
+    val regexPattern = Regex(varKey.split(I18nReference.VARIABLE_MARKER).map { Regex.escape(it) }.joinToString("(.*?)"))
+
+    companion object {
+        val translationFunctions = ImmutableList.of(
```

Comment:
I'd prefer to stick with native Kotlin functions for this, `listOf()` returns an immutable list.